packages/labs/task/src/task.ts

```diff
@@ -215,9 +221,11 @@ export class Task<
         if (error === undefined) {
           this.status = TaskStatus.COMPLETE;
           this._resolveTaskComplete(result as R);
+          this._onComplete?.(result as R);
```

Comment:
If we ever wanted onComplete to be able to change completion status by returning something, we'd want to run this first.